controllers/integrationpipeline/integrationpipeline_adapter_test.go

```diff
@@ -842,16 +845,13 @@ var _ = Describe("Pipeline Adapter", Ordered, func() {
 
 			Expect(statusReporter.Called).To(BeTrue())
 
-			statusAdapter.GetReportersError = errors.New("GetReportersError")
-
-			result, err = adapter.EnsureStatusReported()
-			Expect(result.RequeueRequest && err != nil && err.Error() == "GetReportersError").To(BeTrue())
-
-			statusAdapter.GetReportersError = nil
-			statusReporter.ReportStatusError = errors.New("ReportStatusError")
+			statusAdapter.Reporter = nil
+			if statusAdapter.GetReporter(adapter.pipelineRun) != nil {
+				fmt.Fprintf(GinkgoWriter, "-------test: %v\n", statusAdapter.GetReporter(adapter.pipelineRun))
+			}
 
 			result, err = adapter.EnsureStatusReported()
```

Comment:
I meet one unit test issue [here](https://github.com/redhat-appstudio/integration-service/actions/runs/5949188347/job/16134491871?pr=281#step:18:144),  the reason is `runtime error: invalid memory address or nil pointer dereference` from [here](https://github.com/redhat-appstudio/integration-service/pull/281/files#diff-d2d01b4dc526c804c076232db16ae083ed3381a14d78fc2bbbab8fd4bad59077R164), it looks [this code](https://github.com/redhat-appstudio/integration-service/pull/281/files#diff-d2d01b4dc526c804c076232db16ae083ed3381a14d78fc2bbbab8fd4bad59077R163) doesn't take effect. To debug it, I add [line 849](https://github.com/redhat-appstudio/integration-service/pull/281/files#diff-348a92c0dbf22541261b4c787272ecad09f5f3f277eb2b563ac0c43cecace872R849) and [line 850](https://github.com/redhat-appstudio/integration-service/pull/281/files#diff-348a92c0dbf22541261b4c787272ecad09f5f3f277eb2b563ac0c43cecace872R850)
```
			statusAdapter.Reporter = nil
			if statusAdapter.GetReporter(adapter.pipelineRun) != nil {
				fmt.Fprintf(GinkgoWriter, "-------test: %v\n", statusAdapter.GetReporter(adapter.pipelineRun))
			}
```
It print `-------test: <nil>` [here](https://github.com/redhat-appstudio/integration-service/actions/runs/5949188347/job/16134491871?pr=281#step:18:106). I don't understand why `statusAdapter.GetReporter(adapter.pipelineRun) != nil` is `true` but the value of `statusAdapter.GetReporter(adapter.pipelineRun)` is `nil`. @dirgim Could you help to check it? Thanks!

Reply:
This https://go.dev/doc/faq#nil_error ?

Reply:
https://go.dev/play/p/4eTEC2vUKNN

GetReporter has defined interface as return type, thus nil as interface is not nil value

Reply:
Fixed. Thanks very much!